### PR TITLE
Add major and sha-256 email hash to mailchimp

### PR DESCRIPTION
**Pull request by douglaschan32167**
## Proposed changes
Marketing asked for this information. Tested with my mailchimp account.
Closing #304 
## Types of changes

- [x] I've labeled this PR with appropriate labels (`comp/*` component, bug/feature, documentation and other less important ones).

## Checklist
_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging your code._

- [x] I've linked the proper Github Issue(s) to my PR
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests (unit, integration...etc.) that prove my fix is effective or that my change works
- [x] I have added or updated necessary documentation (if appropriate). You can search, for instance using the [`silver searcher`](https://github.com/ggreer/the_silver_searcher) `ag <STRING>` to find all usages in the repository.
- [x] I have notified the team of any breaking or important changes

### CODE REVIEW

File: backend/api/application_form/models.py [hunk 1]
```diff
@@ -64,6 +64,13 @@ def save(self, *args, **kwargs):
                 how_hear_value = self.referral
                 how_other_value = ''
 
+            mailchimp_client = MailChimp(
+                settings.MAILCHIMP_API_KEY,
+                settings.MAILCHIMP_USERNAME)
+            utf8_email = self.email.lower().encode('utf-8')
```
Comment (iksoosong) [inline backend/api/application_form/models.py:70]: btw, we need to make sure that the hashes match up, so whatever you're applying on here, we need to do on the frontend. we could probably do something like stripping the whitespaces + lowering to normalize the emails.

File: backend/api/application_form/models.py [hunk 2]
```diff
@@ -80,18 +87,15 @@ def save(self, *args, **kwargs):
...
-            mailchimp_client = MailChimp(
-                settings.MAILCHIMP_API_KEY,
-                settings.MAILCHIMP_USERNAME)
-            utf8_email = self.email.lower().encode('utf-8')
-            email_hash = md5(utf8_email)
             # add the new user to the mailchimp list
             mailchimp_client.lists.members.create_or_update(
                 settings.MAILCHIMP_LIST_ID,
-                email_hash.hexdigest(),  # subscriber_hash
+                email_hash,  # subscriber_hash
```
Comment (iksoosong) [inline backend/api/application_form/models.py:98]: why pass in an md5 hash instead of the sha256?

File: backend/api/application_form/models.py [hunk 3]
```diff
@@ -100,7 +104,7 @@ def save(self, *args, **kwargs):
             tag_list = [{'name': tag_name, 'status': 'active'} for tag_name in tags_to_add]
             mailchimp_client.lists.members.tags.update(
                 settings.MAILCHIMP_LIST_ID,
-                email_hash.hexdigest(),
+                email_hash,
```
Comment (iksoosong) [inline backend/api/application_form/models.py:107]: same here

File: backend/api/application_form/models.py (hunk 1 above)
Comment (douglaschan32167) [reply backend/api/application_form/models.py]: I tried it and compared it to the gtm sha256. It matched.

Comment (douglaschan32167) [reply backend/api/application_form/models.py]: At least for the 1 time I tried it.

File: backend/api/application_form/models.py (hunk 2 above)
Comment (douglaschan32167) [reply backend/api/application_form/models.py:99]: I do both. Mailchimp requires MD5, but Dustin also wants to store SHA256. (the MD5 hash was already there; the SHA 256 is a new addition)

Comment (douglaschan32167) [reply backend/api/application_form/models.py:99]: In this case, mailchimp associates users with an MD5 hash, so if I want to check if the user with the given email already exists, I have to give mailchimp the MD5 hash. This is a mailchimp-internal requirement

Comment (douglaschan32167) [reply backend/api/application_form/models.py:99]: Mailchimp is not particularly well-documented, but you can see the api instructions that specify md5 here
https://mailchimp.com/developer/guides/manage-subscribers-with-the-mailchimp-api/

**Comment by douglaschan32167** [comment]
Tried it for 3 more emails. Top is from the python sha256, bottom is what our js produces. Not going to show the actual emails because that's not relevant.
9a440849df85167d4af76189731615d2a23bd05892adea785e100631b9b299e9
9a440849df85167d4af76189731615d2a23bd05892adea785e100631b9b299e9

3abb9d526cff4c9937306c78aa75b91b6c176f33bb08afc62411f3e251aa2b56
3abb9d526cff4c9937306c78aa75b91b6c176f33bb08afc62411f3e251aa2b56

96b250b264d0be4e67c047337e1eba7d8dc21d18800a36abb2259bd1129dc8fe
96b250b264d0be4e67c047337e1eba7d8dc21d18800a36abb2259bd1129dc8fe

**Comment by iksoosong** [comment]
> Tried it for 3 more emails. Top is from the python sha256, bottom is what our js produces. Not going to show the actual emails because that's not relevant.
> 9a440849df85167d4af76189731615d2a23bd05892adea785e100631b9b299e9
> 9a440849df85167d4af76189731615d2a23bd05892adea785e100631b9b299e9
> 
> 3abb9d526cff4c9937306c78aa75b91b6c176f33bb08afc62411f3e251aa2b56
> 3abb9d526cff4c9937306c78aa75b91b6c176f33bb08afc62411f3e251aa2b56
> 
> 96b250b264d0be4e67c047337e1eba7d8dc21d18800a36abb2259bd1129dc8fe
> 96b250b264d0be4e67c047337e1eba7d8dc21d18800a36abb2259bd1129dc8fe

Thanks @douglaschan32167 !!

**Pull request merged**